### ISSUE TITLE
Fails upon trying to copy icon file

### ISSUE THREAD

**Issue by CarVac** [probonopd/linuxdeployqt#136]
I'm trying to package my program, [Filmulator](https://github.com/CarVac/filmulator-gui).

It's a C++ program with a QML UI.

I set up a directory with the executable, .desktop file, icon, and a subdirectory holding the QML files, and ran linuxdeployqt on the .desktop file, and it failed upon trying to copy the icon to where it already was.

TheAssassin on the AppImage IRC told me to make a bug for it, since it's a known issue, and recommended that I put the icon in a `user/share/icons/hicolor/` directory, which at least got past that step.

**Comment by probonopd** [comment]
Which exact error message were you getting?

**Comment by probonopd** [comment]
Is it related to this code?
https://github.com/probonopd/linuxdeployqt/blob/0f2dba46ffa86bd112b3f050b6f9994987ff5970/linuxdeployqt/main.cpp#L237-L320

**Comment by CarVac** [comment]
Yes, it was reaching line 314 there, and saying
`ERROR: Could not copy "/home/carvac/dev/filmulator_appimage/filmulator-gui64.png" to "/home/carvac/dev/filmulator_appimage/filmulator-gui64.png"`.

**Comment by probonopd** [comment]
Strange, if there is a pre-existing icon then it should not attempt to copy:

https://github.com/probonopd/linuxdeployqt/blob/0f2dba46ffa86bd112b3f050b6f9994987ff5970/linuxdeployqt/main.cpp#L305-L317

**Comment by slowphil** [comment]
I encounter this bug too, but in an irreproducible way. I'm buiding the [TeXmacs AppImage on Opensuse Open Buid Service](https://build.opensuse.org/package/show/home:slowphil:texmacs-devel/texmacs), building sometimes fails because of this icon copy error and sometimes it goes through. No changes to the code is involved, it seems to be due to running on different machines in their infrastructure.

I attach the relevant log excerpts of a failure and of a success. The obvious difference between success and failure is that the various icons are found in a different order, resulting a in different icon selected (texmacs-72.png vs texmacs.png). It seems the test for the pre-existing icon is too restrictive in the names it searches.

[excerpt_success.log](https://github.com/probonopd/linuxdeployqt/files/3787733/excerpt_success.log)
[excerpt_fail.log](https://github.com/probonopd/linuxdeployqt/files/3787734/excerpt_fail.log)

**Comment by probonopd** [comment]
`texmacs-72.png` should never get selected, that's the bug then. Can you find out why it happens?

**Comment by slowphil** [comment]
Yeah, the $BUILD_APPDIR is /usr/src/packages/texmacs-0-BuildXXX-x86_64, so every icon found in it contains "64" and matches the test. That's why it's always the last icon that gets selected.

I worked around the issue by manually copying the icon I want beforehand.

**Comment by probonopd** [comment]
Do you have an idea how we could improve the logic in `linuxdeployqt`?
